internal/docstore/docstore.go

```diff
@@ -143,6 +143,11 @@ func (l *ActionList) Get(doc Document, fps ...FieldPath) *ActionList {
 // No field path in mods can be a prefix of another. (It makes no sense
 // to, say, set foo but increment foo.bar.)
 //
+// It is undefined whether updating a sub-field of a non-map field will succeed.
```

Comment:
It kind of sucks to leaves this undefined since that's not very portable. Can we make it always an error?

Reply:
Agreed, but we don't have the document so we can't check.

Reply:
The driver could do an extra `Get` to find out, right? I.e., drivers that natively return an error wouldn't have to do anything, but drivers that don't could do extra work to make it happen. This is a tradeoff between portability/consistency and performance.

Can you add a TODO and/or file a bug to consider this further?

Reply:
Filed #1530.

I'm uncomfortable with the performance cost of a hidden Get.

Here's an idea: `Update` could check against the document it is given. That would work if the document is the result of a `Get`, which I think it often would be: read-modify-write cycles are typical, especially since they provide an alternative to transactions via optimistic locking (that's what the revision field is about). If the document given to `Update` didn't have the field (a blind update), then we'd just make the RPC rather than doing a Get.